### DISCUSSION TITLE
Upgrading .NET 7 Blazor WASM App to .NET 8 Web App with Service Worker for Offline Support

### DISCUSSION THREAD

**Discussion by Kyemate** [dotnet/aspnetcore#52250] [Q&A]
Hello,

We're about to upgrade our Blazor WebAssembly application from .NET 7 to .NET 8 Web App. A critical aspect of our application is its ability to function offline, particularly for certain pages, as it's often used in areas with poor internet connectivity. We achieve this through a service worker.

However, there doesn't seem to be any information about offline support or service worker compatibility in .NET 8 Web Apps?

Thanks!

**Comment by martadon** [comment]
I'm facing the exact same issue. What's puzzling for me is that when I add an empty sw-registrator.js under client-wwwroot, it allows me to navigate between pages, but refreshing doesn't work. It's very strange—I haven't called the registrator, and there's nothing in it. Just having it there seems to activate something. No service-worker.js is present, and none gets installed, yet I can navigate between pages in a new Blazor Web App template (such as between home and counter).

I will follow this thread because I also need a solution for some kind of offline support in my web application (We have SQLite in the browser local storage for offline data that we sync to the server when online).
